packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreDuplicateRoutes.spec.ts

```diff
@@ -0,0 +1,154 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import {
+    gcBlobPrefix,
+    gcTreeKey,
+    IAckedSummary,
+    IContainerRuntimeOptions,
+} from "@fluidframework/container-runtime";
+import { IContainerRuntimeBase, IGarbageCollectionState } from "@fluidframework/runtime-definitions";
+import { ISummaryBlob, ISummaryTree } from "@fluidframework/protocol-definitions";
+import { SharedMap } from "@fluidframework/map";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import { TelemetryNullLogger } from "@fluidframework/common-utils";
+import { TestDataObject, loadSummarizer, submitAndAckSummary } from "../mockSummarizerClient";
+import { mockConfigProvider } from "./mockConfigProivder";
+import { wrapDocumentServiceFactory } from "./gcDriverWrappers";
+
+/**
+ * Validates this scenario: When a datastore is aliased that it is considered a root datastore and always referenced
+ */
+describeNoCompat("GC Data Store Duplicates", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            disableSummaries: true,
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const settings = {
+        "Fluid.GarbageCollection.LogUnknownOutboundRoutes": "true",
+        "Fluid.GarbageCollection.WriteDataAtRoot": "true",
+    };
+    const configProvider = mockConfigProvider(settings);
+    settings["Fluid.GarbageCollection.UseDataStoreAliasing"] = "false";
```

Comment:
This can be moved to where settings is initialized.

Reply:
DataStoreAliasing is set to true by default when we create the mockConfigProvider, so we have to set it to false afterwards. I don't think we care about this case anyway as this issue happens at the DDS level more than the DS level.